python/pyarrow/tests/test_dataset.py

```diff
@@ -3269,8 +3308,19 @@ def test_write_dataset_parquet(tempdir):
 
     # using default "parquet" format string
 
+    files_correct_metadata = 0
+
+    def file_visitor(written_file):
+        nonlocal files_correct_metadata
+        if (written_file.metadata is not None and
+                written_file.metadata.num_columns == 3):
+            files_correct_metadata += 1
```

Comment:
Can you maybe move this into a separate test? (it makes reading the current test for basic functionality a bit complicated IMO)

Reply:
I moved this check into its own test.